src/main/java/seedu/address/logic/commands/AddGradeCommand.java

```diff
@@ -64,7 +63,7 @@ public CommandResult execute(Model model) throws CommandException {
         try {
             Person updatedPerson = personToEdit.addGrade(this.toAdd);
             model.setPerson(personToEdit, updatedPerson);
-            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+            model.updateFilteredPersonList(model.getCurrentPersonFilter());
```

Comment:
Is there a reason for this? It's essentially reassigning the same predicate.

I tried removing this line from `EditCommand` and the behavior seems identical.

Reply:
You're right. I was under the impression that we had to call update to display any changes made to the list, but this is not the case. I've removed the unnecessary commands!